### ISSUE TITLE
[🐛BUG] SASRec 無法執行預設資料集 'ml-100k' 以外的所有資料集

### ISSUE THREAD

**Issue by ChouChou-Justin** [RUCAIBox/RecBole#929]
在嘗試 SASRec 的過程中會不斷出現各種 package 版本不合的報錯，進行各種更新後走向了一個死胡同。

    from recbole.config import Config
    from recbole.data import create_dataset, data_preparation
    from recbole.data.dataloader.abstract_dataloader import AbstractDataLoader
    from recbole.data.dataloader.sequential_dataloader import SequentialDataLoader
    from recbole.model.sequential_recommender import *
    from recbole.trainer import Trainer
    
    parameter = {
        'USER_ID_FIELD': 'user_id',
        'ITEM_ID_FIELD': 'item_id',
        'RATING_FIELD': 'rating', 
        'TIME_FIELD': 'timestamp', 
        'LABEL_FIELD': 'label', 
        'NEG_PREFIX': 'neg_', 
        'ITEM_LIST_LENGTH_FIELD': 'item_length', 
        'LIST_SUFFIX': '_list', 
        'MAX_ITEM_LIST_LENGTH': 50, 
        'POSITION_FIELD': 'position_id', 
        'HEAD_ENTITY_ID_FIELD': 'head_id', 
        'TAIL_ENTITY_ID_FIELD': 'tail_id', 
        'RELATION_ID_FIELD': 'relation_id', 
        'ENTITY_ID_FIELD': 'entity_id', 
        'load_col': {
            'inter': ['user_id', 'item_id', 'rating', 'timestamp'], 
            'user': ['user_id', 'age', 'gender', 'occupation'], 
            'item': ['item_id', 'release_year']
        }, 
        'dataset': 'ml-1m',
        'eval_setting': 'TO_LS, uni100', 
        'valid_metric': 'AUC', 
        'topk': [5, 10, 20],
        'split_to': 5,
        'split_ratio': [0.8, 0.1, 0.1],
        'group_by_user': True,
        'loss_type': 'CE', #Related to training_neg_sample_num in Training Setting
        'training_neg_sample_num': 0,
        'learning_rate choice': [0.01,0.005,0.001,0.0005,0.0001],
        'attn_dropout_prob choice': [0.2, 0.5],
        'hidden_dropout_prob choice': [0.2, 0.5],
        'hidden_act': 'gelu', 
        'n_heads choice': [1, 2],
        'n_layers choice': [1,2,3], 
        'layer_norm_eps': 1e-12, 
        'initializer_range': 0.02, 
        'normalize_all': True
    }
    
    if __name__ == '__main__':
        model_name = 'SASRec'
    
        config = Config(model=model_name, dataset='ml-1m', config_dict=parameter)
        dataset = create_dataset(config)
        train_data, valid_data, test_data = data_preparation(config, dataset)
    
        model = SASRec(config=config, dataset=train_data).to(config['device'])
        trainer = Trainer(config, model)
        best_valid_score, best_valid_result = trainer.fit(train_data, valid_data)
    
        test_result = trainer.evaluate(test_data)
        print('==================' + model_name + '=================')
        for k in parameter['topk']:
            metric = ['recall@' + str(k), 'mrr@' + str(k), 'ndcg@' + str(k), 'hit@' + str(k), 'precision@' + str(k)]
            line = ''
            for m in metric:
                line += m + ':{}'.format(test_result.get(m)) + '\t'
            print(line)

以上 code 會導致 keras 2.3.1 版本相關報錯，但這個報錯出現之前，有另一報錯要求更新至 keras 2.2 以上之版本。
![image](https://user-images.githubusercontent.com/32087670/129467747-11706bc5-847f-4929-bb7a-3db34979cad8.png)

請問是版本相容問題亦或是在 code 中有錯誤導致一連串的問題產生？

- 操作系统: Windows
- RecBole 版本 0.2.0
- Python 版本 3.6.4
- PyTorch 版本 1.1.7
- cudatoolkit 版本 11.4

**Comment by 2017pxy** [comment]
@ChouChou-Justin 你好，根据你给出的版本信息，你的pytorch版本过低，RecBole要求的pytorch版本为1.7.0及以上，关于其他环境依赖的版本要求，请你按照我们的[requirements.txt](https://github.com/RUCAIBox/RecBole/blob/master/requirements.txt)进行配置和升级，然后再尝试运行一下。
